src/test/java/edu/hm/hafner/analysis/parser/JavacParserTest.java

```diff
@@ -274,5 +274,13 @@ private void assertThatWarningIsAtLine(final Issue annotation, final int lineNum
                             "/home/hudson/hudson/data/jobs/Hudson main/workspace/remoting/src/test/java/hudson/remoting/BinarySafeStreamTest.java");
         }
     }
+
+    @Test
+    void shouldParseJavaWarningsInMavenCompilerPlugin(){
+        Report warnings = parse("tracker_issue63346.log");
+        assertThat(warnings).hasSize(4);
+        assertThat(warnings.get(0)).hasSeverity(Severity.WARNING_NORMAL);
```

Comment:
Hey @KalleOlaviNiemitalo 

I have added assert to JavacParser. It asserts all 5 warnings which is the bug. 

But the JavaDocParserTest does not parse a single warning. Ideally it should recognize 2 of them. The assertion fail when I assert `hasSize(2)`. It is expecting size 0 now.

Reply:
Where I should fix the JavacParserTest bug? 

I can work on this, right? Because the bug is exposed now?

Reply:
Hey @KalleOlaviNiemitalo 

I have moved the test from the JavacParserTest to AntJavacParserTest. 

As you have mentioned it only parsed lines without parenthesis for line numbers.

The required fix is, to change the `ANT_JAVAC_WARNING_PATTERN`
https://github.com/jenkinsci/analysis-model/blob/b5551d896fc60f0b5886be365695aeff84be4614/src/main/java/edu/hm/hafner/analysis/parser/AntJavacParser.java#L24-L28

to parse warnings that have parenthesis in line number?

And `JavaDocParserTest` should parse warnings without parenthesis in line number? For now, my JavaDocParserTest does not parse any lines.

Reply:
Seems that the output of the warnings of javac and the javadoc tool changed in recent versions. Sorry about that, I wasn't aware of those problems.

So I fixed the log file for this test and adapted the number of warnings that I am expecting for each of the three parsers. I did not yet change the assertions after the size check, this is something you can add @Jagrutiti. 

The new log is from a recent compile step here on my machine using Java 17. 
I would expect it to have 4 compiler warnings (JavacParser) and 3 java doc warning (JavaDocParser) and 0 AntJavacParser warnings.

Next steps:
- change all three parsers so that the number of warnings is correct.
- adapt the assertions after the size check.

Reply:
Hey @uhafner 

The assertions are not what you expect.

```
[ERROR] Failures: 
[ERROR]   AntJavacParserTest.shouldParseJavaWarningsInMavenCompilerPlugin:231 
Expecting size of:
  <- (-): 2 issues (0 duplicates)>
to be:
  <0>
but was:
  <2>
[ERROR]   JavaDocParserTest.shouldParseJavaDocsWarningsWithMavenSourcePlugin:318 
Expecting size of:
  <- (-): 1 issues (0 duplicates)>
[ERROR]

```

`JavacParser `has 4 warnings as expected.

But, 
`JavaDocParserTest` has 1 warning size.
`AntJavacParserTest` has 2 warnings.

How shall I proceed now?

Reply:
Hey @KalleOlaviNiemitalo , @uhafner 

I have updated the test so that they pass. 

Asserted them with fileName for a better understanding of what the test case expects.

>I would expect it to have 4 compiler warnings (JavacParser) and 3 java doc warning (JavaDocParser) and 0 AntJavacParser warnings.

According to this. I need the bug is in JavaDocParser, and AntJavacParser?

Reply:
The reasoning of TDD is to create a test that *fails* and then change the code so that the red test will turn green. So you need to restore my original assertions, fix JavaDocParser, and AntJavacParser and then the tests should all get green. (And the existing test should stay green).

Reply:
Hey @uhafner 

I have rectified the two test cases to fail.

I understand the following reasons for the failure:

-  AntJavacParser shouldn't parse the JavaDoc warnings. 
-  JavaDocParser should parse all its 3 warnings. But is parses only one.

I was trying to find the difference between the warnings when  I found that
JavaDocParser test parsed the line with `@` before the last word all the test cases turn green.

The following is the sample input that turns them green:
```
[WARNING] Javadoc Warnings
[WARNING] /Users/hafner/git/warnings-ng-plugin-devenv/codingstyle/src/main/java/edu/hm/hafner/util/FilteredLog.java:135: warning: no @return
[WARNING] public int size() {
[WARNING] ^
[WARNING] /Users/hafner/git/warnings-ng-plugin-devenv/codingstyle/src/main/java/edu/hm/hafner/util/LookaheadStream.java:49: warning: no @comment
[WARNING] public String getFileName() {
[WARNING] ^
[WARNING] /Users/hafner/git/warnings-ng-plugin-devenv/codingstyle/src/main/java/edu/hm/hafner/util/ResourceExtractor.java:70: warning: no @comment
[WARNING] public String getResourcePath() {
[WARNING] ^
[WARNING] 3 warnings
```

While in the current input the lines are without `@`:

```

[WARNING] /Users/hafner/git/warnings-ng-plugin-devenv/codingstyle/src/main/java/edu/hm/hafner/util/LookaheadStream.java:49: warning: no comment

[WARNING] /Users/hafner/git/warnings-ng-plugin-devenv/codingstyle/src/main/java/edu/hm/hafner/util/ResourceExtractor.java:70: warning: no comment

```

Does that means
JavaDocParser regex should look for `@`?

And AntJavacParser,
regex should ignore `@`?

or should I write a method for these fixes?

What is the ideal solution here? 
Is my speculation correct?

Reply:
The best way would be to create a new subclass `AbstractMavenLogParser` for `MavenConsoleParser`, `AntJavacParser`, `JavaDocParser`, `JavacParser`: this subclass should use the code in `MavenConsoleParser` to detect the plugin and goal. 

Then:  
- `AntJavacParser` should ignore all warnings, if a maven goal is set (in `isLineInteresting`). Otherwise proceed as normal. 
- `JavaDocParser`, `JavacParser` should ignore warnings, that are from a different maven plugin.
- `MavenConsoleParser` should work unchanged, just the  goal and plugin detection will be from the parent class